From 571790b1bc6ea821ac3bbfc61257eb07d83ec39f Mon Sep 17 00:00:00 2001
From: Paul Naoumenko <paulx.naoumenko@intel.com>
Date: Mon, 14 Oct 2013 20:01:27 +0200
Subject: [PORT FROM MAIN] wifi: Enable Wifi on SaltBay LnP

BZ: 121655

Add support for Intel connectivity in WPA supplicant

Change-Id: I0be2abc012a2d1a6d7f0da9a7f56cac045d23ad8
Orig-Change-Id: Ie2c9aa25bfda65baada3a2ef2bf4ce7b4cbe8454
Category: device enablement
Domain: CWS.WIFI-Intel_LNP
Origin: upstream
Upstream-Candidate: no, proprietary
Signed-off-by: Mick CORNUT <mick.cornut@intel.com>
Signed-off-by: Jonathan DE CESCO <jonathan.de.cesco@intel.com>
---
 Android.mk                   |    4 ++++
 src/drivers/driver_nl80211.c |    6 +++---
 wpa_supplicant/Android.mk    |    7 ++++++-
 wpa_supplicant/Makefile      |    4 ++++
 4 files changed, 17 insertions(+), 4 deletions(-)

diff --git a/Android.mk b/Android.mk
index 72bbcd3..46a101f 100644
--- a/Android.mk
+++ b/Android.mk
@@ -18,3 +18,7 @@ endif
 ifeq ($(strip $(BOARD_WLAN_DEVICE)),wifi-bigcore)
 include $(LOCAL_DIR)/wpa_supplicant/Android.mk
 endif
+
+ifeq ($(strip $(BOARD_WLAN_DEVICE)),iwlwifi)
+include $(LOCAL_DIR)/wpa_supplicant/Android.mk
+endif
diff --git a/src/drivers/driver_nl80211.c b/src/drivers/driver_nl80211.c
index e8f7dc9..f9765fa 100644
--- a/src/drivers/driver_nl80211.c
+++ b/src/drivers/driver_nl80211.c
@@ -10368,7 +10368,7 @@ static int nl80211_set_p2p_powersave(void *priv, int legacy_ps, int opp_ps,
 		   "opp_ps=%d ctwindow=%d)", legacy_ps, opp_ps, ctwindow);
 
 	if (opp_ps != -1 || ctwindow != -1)
-#ifdef ANDROID_P2P
+#if defined(ANDROID_P2P) && !defined(CONFIG_NO_PRIVATE_LIB)
 		wpa_driver_set_p2p_ps(priv, legacy_ps, opp_ps, ctwindow);
 #else
 		return -1; /* Not yet supported */
@@ -10871,12 +10871,12 @@ const struct wpa_driver_ops wpa_driver_nl80211_ops = {
 	.update_ft_ies = wpa_driver_nl80211_update_ft_ies,
 	.get_mac_addr = wpa_driver_nl80211_get_macaddr,
 	.get_survey = wpa_driver_nl80211_get_survey,
-#ifdef ANDROID_P2P
+#if defined(ANDROID_P2P) && !defined(CONFIG_NO_PRIVATE_LIB)
 	.set_noa = wpa_driver_set_p2p_noa,
 	.get_noa = wpa_driver_get_p2p_noa,
 	.set_ap_wps_ie = wpa_driver_set_ap_wps_p2p_ie,
 #endif
-#if defined(ANDROID) && !defined(NO_PRIVATE_LIB)
+#if defined(ANDROID) && !defined(CONFIG_NO_PRIVATE_LIB)
 	.driver_cmd = wpa_driver_nl80211_driver_cmd,
 #endif
 #if defined(ANDROID) && defined(ANDROID_BIGCORE)
diff --git a/wpa_supplicant/Android.mk b/wpa_supplicant/Android.mk
index 88f301b..20a5c86 100644
--- a/wpa_supplicant/Android.mk
+++ b/wpa_supplicant/Android.mk
@@ -38,6 +38,11 @@ ifeq ($(BOARD_WLAN_DEVICE), wifi-bigcore)
 L_CFLAGS += -DANDROID_BIGCORE
 endif
 
+ifeq ($(BOARD_WLAN_DEVICE), iwlwifi)
+L_CFLAGS += -DANDROID_QCOM_WCN
+L_CFLAGS += -DANDROID_P2P
+endif
+
 ifeq ($(BOARD_WLAN_DEVICE), qcwcn)
 L_CFLAGS += -DANDROID_P2P
 endif
@@ -1560,7 +1565,7 @@ endif
 ifneq ($(BOARD_WPA_SUPPLICANT_PRIVATE_LIB),)
 LOCAL_STATIC_LIBRARIES += $(BOARD_WPA_SUPPLICANT_PRIVATE_LIB)
 else
-L_CFLAGS += -DNO_PRIVATE_LIB
+L_CFLAGS += -DCONFIG_NO_PRIVATE_LIB
 endif
 LOCAL_SHARED_LIBRARIES := libc libcutils liblog
 ifeq ($(CONFIG_TLS), openssl)
diff --git a/wpa_supplicant/Makefile b/wpa_supplicant/Makefile
index 5698619..3b09339 100644
--- a/wpa_supplicant/Makefile
+++ b/wpa_supplicant/Makefile
@@ -1510,6 +1510,10 @@ EXTRALIBS += WbemUuid.Lib
 endif
 endif
 
+ifdef CONFIG_NO_PRIVATE_LIB
+CFLAGS += -DCONFIG_NO_PRIVATE_LIB
+endif
+
 ifndef LDO
 LDO=$(CC)
 endif
-- 
1.7.4.1

